### ISSUE TITLE
Open With

### ISSUE THREAD

**Issue by Beep6581** [BestImageViewer/geeqie#535]
Screenshots taken using 39cbcfb5, tested again using f882d3ca

I need to be able to select a few photos and "Open with" some program or script. The closest thing I can find to that is Edit > External Editors. There I see GIMP, Gwenview, Hugin, my script, Inkscape and RawTherapee. None of those options work, they all produce this:
![imgur-2017_11_04-13 28 22](https://user-images.githubusercontent.com/5844619/32405357-0f83a5e0-c164-11e7-9df2-e47a60c1c251.png)

If I right-click on the selected thumbs, I don't see those programs in the context menu. All I see in the "Edit" context menu is "Symlink".

Edit > Preferences > Configure Editors 
![screenshot_20171104_133017](https://user-images.githubusercontent.com/5844619/32405368-5603f880-c164-11e7-93b6-419bfe4b66c3.png)
![screenshot_20171104_133130](https://user-images.githubusercontent.com/5844619/32405379-7d90e372-c164-11e7-8c58-aa11b5323f0b.png)

I tried copying these lines from "Symlink" and pasting them into my Bash script Imgination:
```
# Desktop files that are usable only in Geeqie should be marked like this:
Categories=X-Geeqie;
OnlyShowIn=X-Geeqie;

# Show in menu "File"
X-Geeqie-Menu-Path=FileMenu/FileOpsSection
```
Now Imgination appears under the File menu, but fails to run with the same error, "External command failed".

I as a programmer am unable to figure out how to open a few selected photos with my script or with GIMP or any other program. How is this supposed to work?

**Comment by mowgli** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Hi,

Am Sa den  4. Nov 2017 um 13:40 schrieb Beep6581:
> Screenshots taken using 39cbcfb5, tested again using f882d3ca

Please do so.

> I need to be able to select a few photos and "Open with" some program or script. The closest thing I can find to that is Edit > External Editors. There I see GIMP, Gwenview, Hugin, my script, Inkscape and RawTherapee. None of those options work, they all produce this:
> ![imgur-2017_11_04-13 28 22](https://user-images.githubusercontent.com/5844619/32405357-0f83a5e0-c164-11e7-9df2-e47a60c1c251.png)

I tried to reproduce this problem but failed. (I use gimp to open and it
worked well as expected.)

Regards
   Klaus
- -- 
Klaus Ethgen                                       http://www.ethgen.ch/
pub  4096R/4E20AF1C 2011-05-16            Klaus Ethgen <Klaus@Ethgen.ch>
Fingerprint: 85D4 CA42 952C 949B 1753  62B3 79D0 B06F 4E20 AF1C
-----BEGIN PGP SIGNATURE-----
Comment: Charset: ISO-8859-1

iQGzBAEBCgAdFiEEMWF28vh4/UMJJLQEpnwKsYAZ9qwFAln+IEUACgkQpnwKsYAZ
9qx+yQv/cH2EoSuXwRBBeP6Y4AhT0g2Knl4RCP48rvnjcdsihhEms8Z47ez07dU4
W6TPXfHaFbIeZut7Isr9YSHbQowjRwFtGgBUo2KegtgIip0JCISb2UnirpsLWD+9
xVowM0BvZJTLPRFbTMBsOBoBazMyJbu7r7uI+hiVe48GTFcgFQAdfFQduAWucEgA
7lzoxSjbuCWWhu83N8Rx2jzRCEXzCJIEXTMzs4eAQ2MTmJnYMr3pZofKxYxSMQiJ
JU2cB3x9lesYOVpxLuE+wehN8Z5KV5UC2AAvCyqshOwU5QTdltscxiD7/c2QsAdL
WTmquKS2V64PeQ3jT2TBq3XAL7sWT0DWom0jKQMP/aL8358Rm4l2NLpLK/jPz2/n
+Egrg04WZCVKWYxOw6ZOhWZM9uDRJeMrRtQW5iXvRf//LEvg8Zn9uJ/egR0g1O+H
JwTO/QbXH3mF44R8NmROVAnSTt2k1XruVLsZyWFyKBzoWm5rgEeL+/wu1o2WnggL
4ap+7yt1
=cxxg
-----END PGP SIGNATURE-----

**Comment by caclark** [comment]
> If I right-click on the selected thumbs, I don't see those programs in the context menu.

When loading the right-click menu, the program checks MimeTypes against the file extension of the selected file to decide which items to display -  and clearly doesn't find a match.

Which is why you get "Can't find matching file" when you run off the main menu.

In the case of your desktop file I suggest commenting out the MimeType line.
I have no idea why GIMP etc. should all be blocked.

**Comment by Beep6581** [comment]
That was it!

My original desktop file was in `/usr/share/applications/imgination.desktop` and I found that Geeqie created a copy in `~/.config/geeqie/applications/imgination.desktop`. I deleted the copy (I know I didn't have to) and removed the `MimeType` line from the one in /usr/share, and now it appears and works fine for both raw and non-raw images.

Now invoking Imgination works, thank you.

Since applications appear in the Plugins menu which don't necessarily have a mime type which matches the current image, I see two possible solutions to the confusion:
1. Improve the "External command failed" message. Add something like "None of the plugins have a matching mime type entry. Go to Edit > Configure Plugins and make sure that the required plugin has a MimeType= line which matches the mime type of the image you're trying to open". This makes the problem more clear, but it does not solve anything. So I prefer point 2.
2. Don't check for mime types when the user clicking on any of the items in the Plugins menu. This makes total sense. If I click on a DNG file and click on Plugins > GIMP, Geeqie should not be getting in the way, trying to stop me just because the GIMP desktop file has no mime type for DNG. In this case the user knows better.

Currently I have 5 plugins: 
- GNU Image Manipulation Program
- Gwenview
- Imgination
- Inkscape
- RawTherapee

Of those 5, 4 support opening the raw file I clicked on, and yet Geeqie prevents me from using any of them due to the mime type check. I think this is testament to option 2 being a good solution.

**Comment by Beep6581** [comment]
I ran into this issue again, and had to delete the `MimeType` line. Why? The mime-types are valid.

[rawtherapee.desktop.txt](https://github.com/BestImageViewer/geeqie/files/1800480/rawtherapee.desktop.txt)

**Comment by caclark** [comment]
I set up a desktop file with your data, and for me it works.
Perhaps you could include in your desktop file the line:
X-Geeqie-Verbose=true
The additional output might give some help where the problem is.

**Comment by Beep6581** [comment]
@caclark I appended that line to `/usr/share/applications/rawtherapee.desktop` and made sure that there was no other `rawtherapee.desktop` on my system. I started Geeqie, right-clicked on an `.ARW` file, and did not see RT in the Edit context menu. When I clicked on RT in the Plugins menu, I got the same message as above, with no more info and nothing in the console.

Geeqie `v1.2-458-g`36ddb9db

**Comment by Beep6581** [comment]
P.S. Right-clicking on a PNG file did show RT in the Edit menu and RT ran fine, but ARW, PEF, DNG, etc. none of them worked.

**Comment by caclark** [comment]
Fixed in commit da564ede9 at geeqie.org

**Comment by Beep6581** [comment]
@caclark how do I view the commit? Could you link to it?

**Comment by Beep6581** [comment]
By the way, you can find the latest raw mime types supported by RawTherapee here, as a reference:
https://github.com/Beep6581/RawTherapee/blob/dev/rtdata/icons/mime-types

**Comment by caclark** [comment]
You quote commit id 36ddb in a previous message, so I assume you are compiling from sources.
Go to the folder where you cloned geeqie, and use gitk.
Or on-line, http://geeqie.org/cgi-bin/gitweb.cgi?p=geeqie.git is where the repo is.

**Comment by Beep6581** [comment]
Yes, I meant a web interface for when I don't have access to the git repo, e.g. phone. Thank you.
